### PR TITLE
[consensus] remove narwhal

### PR DESCRIPTION
**Pull request by arun-koshy**
## Description 

Goodbye Narwhal! 

There are some small cleanups that still need to be done in follow up PRs

### CODE REVIEW

**Review by bmwill — APPROVED** [review]

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -18,14 +18,10 @@ CHANGELOG.md
...
-
-# Notify Narwhal changes to interested folks.
-/narwhal/crypto/ @joyqvq @kchalkias
-/narwhal/executor/ @asonnino @akichidis @laura-makdah
-/narwhal/network/ @bmwill
-/narwhal/storage/ @akichidis
-
-/crates/sui-core/src/consensus_adapter.rs @laura-makdah
+# Consensus working group
+/consensus/ @akichidis @mwtian @arun-koshy
```
Comment (mwtian) [inline .github/CODEOWNERS:22]: Can you try @MystenLabs/consensus instead?

**Review by mwtian — APPROVED** [review]

File: .github/CODEOWNERS (hunk 1 above)
Comment (bmwill) [reply .github/CODEOWNERS:22]: would it be possible to migrate the consensus crates into /crates with the rest of the crates at some point?

Comment (mwtian) [reply .github/CODEOWNERS:22]: Probably not. There are still `external-crates/` and `sui-execution`.

**Review by akichidis — APPROVED** [review]
Goodbye old friend! Thanks @arun-koshy for the clean up.

**Pull request merged**